### PR TITLE
SearchableSnapshotsIT.testCacheStats should wait for cache writes to complete before checking stats

### LINKED ISSUE #75922: [CI] SearchableSnapshotsIT testCacheStats failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/jd445wainhixw/tests/:client:rest-high-level:asyncIntegTest/org.elasticsearch.client.SearchableSnapshotsIT/testCacheStats

**Reproduction line:**
`./gradlew ':client:rest-high-level:asyncIntegTest' --tests "org.elasticsearch.client.SearchableSnapshotsIT.testCacheStats" -Dtests.seed=38807247BC375C19 -Dtests.locale=es-HN -Dtests.timezone=Asia/Colombo -Druntime.java=8`

**Applicable branches:**
7.13

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.client.SearchableSnapshotsIT&tests.test=testCacheStats

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a value equal to or greater than <1L>
     but: <0L> was less than <1L>

  at __randomizedtesting.SeedInfo.seed([38807247BC375C19:620452429E251267]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.client.SearchableSnapshotsIT.testCacheStats(SearchableSnapshotsIT.java:141)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

```

### PR DESCRIPTION
**Pull request by tlrx**
This test should wait for cache writes to be completed before checking the cache stats.

Closes #75922

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/SearchableSnapshotsIT.java [hunk 1]
```diff
@@ -145,4 +178,9 @@ public void testCacheStats() throws Exception {
             assertThat(stats.getEvictions(), equalTo(0L));
         }
     }
+
+
+    private static void waitForIdlingSearchableSnapshotsThreadPools() throws Exception {
```
Comment (original-brownbear) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/SearchableSnapshotsIT.java:183]: ? :)

**Review by original-brownbear — APPROVED** [review]
LGTM if you either delete the empty method or move the code there :)

File: client/rest-high-level/src/test/java/org/elasticsearch/client/SearchableSnapshotsIT.java (hunk 1 above)
Comment (tlrx) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/SearchableSnapshotsIT.java]: Broken intellij for two days :(

**Comment by tlrx** [comment]
Thanks Armin!

**Pull request merged**